contracts/oracle/BackedOracleFactory.sol

```diff
@@ -0,0 +1,121 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2021-2023 Backed Finance AG
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/governance/TimelockController.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "./BackedOracle.sol";
+
+/**
+ * @dev
+ * TransparentUpgradeableProxy contract, renamed as BackedOracleProxy.
+ */
+contract BackedOracleProxy is TransparentUpgradeableProxy {
+    constructor(
+        address _logic,
+        address _admin,
+        bytes memory _data
+    ) payable TransparentUpgradeableProxy(_logic, _admin, _data) {}
+}
+
+contract BackedOracleFactory is Ownable {
+    ProxyAdmin public proxyAdmin;
+    TimelockController public timelockController;
+    BackedOracle public implementation;
+
+    event NewOracle(address indexed newOracle);
+    event NewImplementation(address indexed newImplementation);
+
+    /**
+     * @param admin- The address of the account that will be set as owner of the deployed ProxyAdmin and will have the
+     *      timelock admin role for the timelock contract
+     */
+    constructor(address admin, address[] memory timelockWorkers) {
+        require(
+            admin != address(0),
+            "Factory: address should not be 0"
+        );
+
+        implementation = new BackedOracle();
+
+        proxyAdmin = new ProxyAdmin();
+        proxyAdmin.transferOwnership(admin);
+
+        timelockController = new TimelockController(
```

Comment:
Note that in this version of OpenZeppelin, the TimelockController grants itself the TIMELOCK_ADMIN_ROLE by default. This means not only the admin has TIMELOCK_ADMIN_ROLE  privileges, but also every timelockWorkers (with time lock). Not sure if that's intended or not but just raising this in case it isn't.